test/DynamoPublishTests/PublishModelTests.cs

```diff
@@ -42,24 +42,62 @@ public void SetUp()
         }
 
         [Test]
-        public void SendWorkspaceTest()
+        public void WorkspaceDependencies_Collect_ShouldWorkCorrectlyForWorkspaceWithNoDeps()
         {
-            string openPath = Path.Combine(TestDirectory, @"Libraries\DynamoPublishTests\PublishWorkspaceTest.dyn");
+            string openPath = Path.Combine(TestDirectory, @"Libraries\DynamoPublishTests\PublishWorkspaceTestNoDeps.dyn");
             OpenModel(openPath);
 
-            publishModel.Send(CurrentDynamoModel.Workspaces);
+            var hws = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
 
-            Assert.IsNotNull(publishModel.HomeWorkspace);
-            Assert.AreEqual(1, publishModel.CustomNodeWorkspaces.Count);
+            Assert.NotNull(hws, "The current workspace is not a " + typeof(HomeWorkspaceModel).FullName);
+
+            var deps = PublishModel.WorkspaceDependencies.Collect(hws, CurrentDynamoModel.CustomNodeManager);
+
+            Assert.IsNotNull(deps.HomeWorkspace);
+            Assert.AreEqual(0, deps.CustomNodeWorkspaces.Count());
+        }
+
+        [Test]
+        public void WorkspaceDependencies_Collect_ShouldWorkCorrectlyForWorkspaceWithOneDep()
+        {
+            string openPath = Path.Combine(TestDirectory, @"Libraries\DynamoPublishTests\PublishWorkspaceTestOneDep.dyn");
+            OpenModel(openPath);
+
+            var hws = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+
+            Assert.NotNull(hws, "The current workspace is not a " + typeof(HomeWorkspaceModel).FullName);
+
+            var deps = PublishModel.WorkspaceDependencies.Collect(hws, CurrentDynamoModel.CustomNodeManager);
+
+            Assert.IsNotNull(deps.HomeWorkspace);
+            Assert.AreEqual(1, deps.CustomNodeWorkspaces.Count());
+        }
+
+        [Test]
+        public void WorkspaceDependencies_Collect_ShouldWorkCorrectlyForWorkspaceWithNestedDeps()
+        {
+            string openPath = Path.Combine(TestDirectory, @"Libraries\DynamoPublishTests\PublishWorkspaceTestTwoDeps.dyn");
+            OpenModel(openPath);
+
+            var hws = CurrentDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(hws, "The current workspace is not a " + typeof(HomeWorkspaceModel).FullName);
+
+            var deps = PublishModel.WorkspaceDependencies.Collect(hws, CurrentDynamoModel.CustomNodeManager);
+
+            Assert.IsNotNull(deps.HomeWorkspace);
+            Assert.AreEqual(2, deps.CustomNodeWorkspaces.Count());
```

Comment:
These tests ensure that all dependencies for the home workspace are traversed and uploaded.